File: MaWPhotos/src/main/java/us/mikeandwan/photos/ui/photos/ThumbnailLinearLayoutManager.java
package us.mikeandwan.photos.ui.photos;

import android.content.Context;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


// this was needed to properly scroll immediately on resume (like during an orientation change)
public class ThumbnailLinearLayoutManager extends LinearLayoutManager {
    private int _startPosition;


    public ThumbnailLinearLayoutManager(Context context, int orientation, boolean reverseLayout, int startPosition) {
        super(context, orientation, reverseLayout);

        _startPosition = startPosition;
    }


    @Override
    public void onLayoutChildren(final RecyclerView.Recycler recycler, final RecyclerView.State state) {
        super.onLayoutChildren(recycler, state);

        if(_startPosition > 0) {
            scrollToPosition(_startPosition);
        }
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/ui/photos/ThumbnailLinearLayoutManager.kt
package us.mikeandwan.photos.ui.photos

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView.Recycler
import androidx.recyclerview.widget.RecyclerView

// this was needed to properly scroll immediately on resume (like during an orientation change)
class ThumbnailLinearLayoutManager(
    context: Context?,
    orientation: Int,
    reverseLayout: Boolean,
    private val _startPosition: Int
) : LinearLayoutManager(context, orientation, reverseLayout) {
    override fun onLayoutChildren(recycler: Recycler, state: RecyclerView.State) {
        super.onLayoutChildren(recycler, state)
        if (_startPosition > 0) {
            scrollToPosition(_startPosition)
        }
    }
}